provider/core/src/datagen/heap_measure.rs

```diff
@@ -0,0 +1,52 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::buf::{BufferFormat, BufferMarker};
+use crate::prelude::*;
+use yoke::trait_hack::YokeTraitHack;
+
+/// Stats on the heap size needed when attempting to zero-copy-deserialize
+/// a postcard-formatted data struct.
+#[derive(Debug, Copy, Clone, yoke::Yokeable, Default)]
+pub struct HeapStats {
+    // Total bytes allocated during deserialization
+    pub total_bytes_allocated: u64,
+    // Total bytes allocated during deserialization that have not yet been freed
+    pub final_bytes_needed: usize,
```

Comment:
Suggestion (optional):
```suggestion
    pub net_bytes_allocated: usize,
```

Reply:
-1, I think `final` is clearer than `net`. Maybe `intermediate` (`total-final`) and `final`?

Reply:
Yeah, I agree with "net"

Reply:
`final` is the final number of bytes that are allocated for this struct. `intermediate` is the number of additional bytes that were allocated during deserialization but then freed again. The fact that we computed these numbers by looking at total memory usage before and after should not have any impact on the naming.